src/tests/rust_fmt_argument_max_padding.rs

```diff
@@ -0,0 +1,157 @@
+//! These tests exist to validate padding behavior when using dynamic widths in
+//! format strings.  As of rust-lang/rust#136932 (part of #99012), the Rust
+//! standard library now restricts width and precision fields in format strings
+//! to `u16::MAX`, to improve memory layout and prevent silent truncation bugs
+//! on cross-compilation targets. This change caused previously valid dynamic
+//! width values exceeding `u16::MAX` to panic at runtime. (See
+//! rust-lang/rust#136932).
+//!
+//! These test cases specifically target edge conditions revealed by fuzzing
+//! strftime-ruby, ensuring we handle excessively large padding values without
+//! panicking, aligning behavior closely with CRuby's `Time#strftime`
+//! specification and the limitations described in rust-lang/rust#136932.
+//!
+//! Reference:
+//!
+//! - <https://github.com/rust-lang/rust/pull/136932>
+//! - <https://github.com/rust-lang/rust/pull/136932#issuecomment-2739434542>
+
+#![allow(clippy::unreadable_literal)]
+
+#[cfg(feature = "alloc")]
+use {
+    crate::format::TimeFormatter,
+    alloc::{format, vec},
+};
+
+use crate::Error;
+
+use super::{check_all, get_format_err, get_format_err_bytes, MockTime};
+
+#[test]
+fn test_larger_than_int_max_formats_are_returned_verbatim() {
+    let times = [
+        MockTime::new(1970, 1, 1, 0, 0, 0, 0, 4, 1, 0, false, 0, ""),
+        MockTime::new(-1970, 1, 1, 0, 0, 0, 0, 4, 1, 0, false, 0, ""),
+    ];
+
+    for format in [
+        "%100000000000000000000c",
```

Comment:
might be worth adding a test for a specifier width that exceeds `i64::MAX` as well:

```console
[3.3.6] > x = 10000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000
=>
10000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000...
[3.3.6] > x > (2 ** 64)
=> true
[3.3.6] > Time.now.strftime "%#{x}c"
=> "%10000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000c"
```

```suggestion
        "%10000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000c",
        "%100000000000000000000c",
```

Reply:
The first value is already 1e20, which is greater than i64::MAX.